### ISSUE TITLE
[Go] Add documentation for Go data plane SDK generation

### ISSUE THREAD

**Issue by maririos** [Azure/azure-sdk-tools#6944]
With the move to our new doc site, there is no information about how a team can generate a Go data plane sdk. 
Instructions will need to be added:

- [x] (generate): https://eng.ms/docs/products/azure-developer-experience/develop/sdk-generate?tabs=dataplane
- [x] (test): https://eng.ms/docs/products/azure-developer-experience/develop/sdk-test?tabs=dataplane
- [ ] (samples): https://eng.ms/docs/products/azure-developer-experience/develop/sdk-samples?tabs=dataplane

**Comment by maririos** [comment]
PR that covers generate SDK and tests: https://dev.azure.com/azure-sdk/internal/_git/azure-sdk-docs-eng.ms/pullrequest/846

**Comment by maririos** [comment]
> Generate

https://eng.ms/docs/products/azure-developer-experience/develop/sdk-generate

> Test

https://eng.ms/docs/products/azure-developer-experience/develop/sdk-test?tabs=dataplane

**Comment by maririos** [comment]
@jhendrixMSFT do you have an estimate of when the sample information could be added to the documentation?

**Comment by jhendrixMSFT** [comment]
We do have [this](https://github.com/Azure/azure-sdk-for-go/blob/main/documentation/developer_setup.md) which the WebPubSub folks used to on-board their data-plane SDK. Does this work?

**Comment by maririos** [comment]
Does the Go SDK need to have samples in order to ship? are the samples handled in a specific way that will be good to add in the document you linked?

**Comment by jhendrixMSFT** [comment]
Yes, I've opened https://github.com/Azure/azure-sdk-for-go/pull/22084 to add a section on testable examples.

**Comment by maririos** [comment]
Thank you!

**Comment by RickWinter** [comment]
We believe this is complete. Is there anything else left on this?
